### PR TITLE
Type conversion to isinstance modernization.

### PR DESCRIPTION
**Pull request by caterpillarAttack**
This is roughly 20% faster and doesnt rely on an equality check under the hood. https://www.pylenin.com/blogs/type-vs-isinstance/#type-vs-isinstance

### CODE REVIEW

File: python/vtool/util.py [hunk 1]
```diff
@@ -837,19 +837,19 @@ def convert_to_sequence(variable, sequence_type = list):
         list, tuple: Returns list or tuple depending on the sequence_type.
     """
     
-    if type(variable) == sequence_type:
+    if isinstance(variable, sequence_type):
         return variable
         
-    if type(variable) == list and sequence_type == tuple:
+    if isinstance(variable, list) and sequence_type == tuple:
```
Comment (caterpillarAttack) [inline python/vtool/util.py:843]: use is here

File: python/vtool/util.py [hunk 2]
```diff
@@ -837,19 +837,19 @@ def convert_to_sequence(variable, sequence_type = list):
...
         variable = tuple(variable)
         return variable
     
-    if type(variable) == tuple and sequence_type == list:
+    if isinstance(variable, tuple) and sequence_type == list:
```
Comment (caterpillarAttack) [inline python/vtool/util.py:847]: use is here

File: python/vtool/process_manager/ui_options.py [hunk 3]
```diff
@@ -428,7 +428,7 @@ def _find_group_widget(self, name, find_last = False):
                     
                     widget_type = type(widget)
                     
-                    if widget_type == ProcessOptionGroup or widget_type == ProcessReferenceGroup:
+                    if widget_type is ProcessOptionGroup or widget_type is ProcessReferenceGroup:
```
Comment (caterpillarAttack) [inline python/vtool/process_manager/ui_options.py:431]: use isinstance

File: python/vtool/process_manager/ui_options.py [hunk 4]
```diff
@@ -583,7 +583,7 @@ def _write_options(self, clear = True):
                 
                 self.process_inst.add_option(name, value, None, widget_type)
             
-            if type(self) == ProcessReferenceGroup:
+            if type(self) is ProcessReferenceGroup:
```
Comment (caterpillarAttack) [inline python/vtool/process_manager/ui_options.py:586]: use isinstance

File: python/vtool/process_manager/ui_options.py (hunk 3 above)
Comment (caterpillarAttack) [reply python/vtool/process_manager/ui_options.py:431]: nevermind, its good as is.

File: python/vtool/process_manager/ui_options.py (hunk 4 above)
Comment (caterpillarAttack) [reply python/vtool/process_manager/ui_options.py:586]: nevermind, its good as is.

**Comment by louisVottero** [comment]
This is doing something to skin weight import, will try to pinpoint it

File: python/vtool/maya_lib/api.py [hunk 5]
```diff
@@ -1452,427 +1452,426 @@ def multiply_matrix(matrix1, matrix2):
...
-            
+
             index = skin_fn.indexForInfluenceObject(influence_dag_paths[inc])
             influence_array.append(index)
-          
-    
+
+
     weight_array = None
-    if type(weights) == type(om.MDoubleArray()):
```
Comment (louisVottero) [inline python/vtool/maya_lib/api.py:1837]: V: Error!	Traceback (most recent call last):	
V:		  File "E:/dev/git/vtool/python\vtool\util.py", line 76, in wrapper	
V:		    return_value = function(*args, **kwargs)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1673, in import_data	
V:		    self._import_maya_data(filepath, selection)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1323, in _import_maya_data	
V:		    result = self.import_skin_weights(folder_path, mesh, first = first)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1568, in import_skin_weights	
V:		    maya_lib.api.set_skin_weights(skin_cluster, weight_array, 0)	
V:		  File "E:/dev/git/vtool/python\vtool\maya_lib\api.py", line 1837, in set_skin_weights	
V:		    if isinstance(weights, om.MDoubleArray()):	
V:		TypeError: isinstance() arg 2 must be a type or tuple of types

Comment (louisVottero) [reply python/vtool/maya_lib/api.py:1837]: The error above I get from Maya when running the human body template
Maybe isinstance doesn't work on om.MDoubleArray()

**Review by louisVottero — CHANGES_REQUESTED** [review]

**Review by louisVottero — APPROVED** [review]

**Pull request merged**